src/components/TransactionsInterpreter/TransactionsInterpreter.ts

```diff
@@ -0,0 +1,117 @@
+import { getTransactionDirection } from './helpers/getTransactionDirection';
+import { NUMBER_OF_CHARACTERS_FOR_SMART_CONTRACT_ADDRESS } from 'constants/transaction-interpreter';
+
+export type DenominationConfig = {
+  egldLabel?: string;
+  decimals?: number;
+  denomination?: number;
+  showLastNonZeroDecimal?: boolean;
+  showLabel?: boolean;
+  token?: string;
+};
+
+export function parseTransactions(
+  transactions: UITransactionType[],
+  address: string,
+  denominationConfig: DenominationConfig = {
+    egldLabel: 'EGLD'
+  },
+  numInitCharactersForScAddress: number = NUMBER_OF_CHARACTERS_FOR_SMART_CONTRACT_ADDRESS
+): ExtendedTransactionType[] {
+  return transactions.map((transaction) => {
+    return processTransaction(
+      transaction,
+      address,
+      denominationConfig,
+      numInitCharactersForScAddress
+    );
+  });
+}
+
+export function processTransaction(
+  transaction: UITransactionType,
+  address: string,
+  denominationConfig: DenominationConfig = {
+    egldLabel: 'EGLD'
+  },
+  numInitCharactersForScAddress: number = NUMBER_OF_CHARACTERS_FOR_SMART_CONTRACT_ADDRESS
+): ExtendedTransactionType {
+  const tokenIdentifier =
+    transaction.tokenIdentifier ??
+    (getTokenFromData(transaction.data).tokenId || getEgldLabel());
+
+  let receiver = transaction.receiver;
+  if (transaction.action && transaction.action.arguments?.receiver) {
+    receiver = transaction.action.arguments.receiver;
+  }
+
+  const receiverAssets =
+    transaction.receiver === receiver ? transaction.receiverAssets : undefined;
+
+  const direction = getTransactionDirection(address, transaction, receiver);
+  const method = getTransactionMethod(transaction);
+  const transactionTokens: TokenArgumentType[] = getTransactionTokens(
+    transaction
+  );
+
+  const denominatedValue = getDenominatedValue(transaction, denominationConfig);
+  const fullDenominatedValue = getDenominatedValue(transaction, {
+    ...denominationConfig,
+    showLastNonZeroDecimal: true
+  });
+
+  const senderLink = getNetworkLink(
+    urlBuilder.accountDetails(transaction.sender)
+  );
+  const receiverLink = getNetworkLink(urlBuilder.accountDetails(receiver));
+  const senderShardLink = getNetworkLink(
+    urlBuilder.senderShard(transaction.senderShard)
+  );
+  const receiverShardLink = getNetworkLink(
+    urlBuilder.receiverShard(transaction.receiverShard)
+  );
+  const transactionLink = getNetworkLink(
+    `/transactions/${
+      transaction.originalTxHash
+        ? `${transaction.originalTxHash}#${transaction.txHash}`
+        : transaction.txHash
+    }`
+  );
+
+  const shortTimeAgo = timeRemaining(transaction.timestamp, true);
+  const longTimeAgo = timeRemaining(transaction.timestamp, false);
+
+  // TODO create getTokenDetails utils function and compute lockedAccountName property in order to use it inside the LockedTokenAddressIcon component
+
+  return {
+    ...transaction,
+    tokenIdentifier,
+    receiver,
```

Comment:
Let's group these into smaller objects, so they have some kind of structure. {links, transactionDetails, dateTime} etc